### DISCUSSION TITLE
New ISBN Search Library

### DISCUSSION THREAD

**Discussion by NetCzech** [textbrowser/biblioteq#109] [Ideas]
Hi,
I'm a new user of the program. Could I ask to add the [WorldCat](https://www.worldcat.org) library to the ISBN book search options?  I'm from the Czech Republic and I have a problem with searching for czech literature.

Thank you very much

**Comment by textbrowser** [comment]
The request exists but is not implemented. So, no. :)

https://github.com/textbrowser/biblioteq/discussions/89

**Comment by textbrowser** [comment]
https://www.loc.gov/cgi-bin/zgate?ACTION=INIT&FORM_HOST_PORT=/prod/www/data/z3950/nlczechrepublic.html,aleph.nkp.cz,9991&CI=160941

**Comment by textbrowser** [comment]
https://aleph.nkp.cz/F/5M6S3TMFEHJ6E2CAHY415X4GQKCJQRSTTL6L1BEI9BIVQCDRBX-31989?func=file&file_name=base-list

**Comment by textbrowser** [comment]
![Screenshot_20210925_211139](https://user-images.githubusercontent.com/10701156/134789762-e15c05ac-e0ef-47f5-b62d-f0bf853871f8.png)

**Comment by textbrowser** [comment]
![Screenshot_20210925_211832](https://user-images.githubusercontent.com/10701156/134789918-addcee83-ccfa-4497-b97c-44682e8590c5.png)

**Comment by NetCzech** [comment]
That looks perfect. Please, how can I add the option to the program?

**Comment by textbrowser** [comment]
BiblioteQ is open source and thus everything is open. :P Discover it and learn it.

In short, see biblioteq.conf. Learn about Z39.50. I did a bunch of work locating the National Czech Library and its Z39.50 information as I could not readily find it on the library's site. So yah, the difficult work was solved for you for free. :)

**Reply by NetCzech** [reply]
Oh, thank you for your help ;)

**Comment by NetCzech** [comment]
I tried to add a record for the National Library of Czech Republic:

```
database_name =
format = "skc-utf"
hostname = aleph.nkp.cz
name = National Library of Czech Republic
password =
port = 9991
proxy_host =
proxy_port =
record_syntax = "USMARC"
username =
```

but, I found that my Z39.50 search does not work.

![01_BiblioteQ](https://user-images.githubusercontent.com/76791894/134938278-ae7179c6-314a-4ceb-a182-fa18ab1e43b2.png)

![02_BiblioteQ](https://user-images.githubusercontent.com/76791894/134938294-e626e15e-c3f6-4fe6-846a-a73136fc4a56.png)

OpenLibrary and SRU work well. The error occurs with all Z39.50 libraries.

**Comment by textbrowser** [comment]
Please paste the contents of the Error Log.

**Comment by NetCzech** [comment]
It's a right log?

ErrorLog:
2021/09/27 17:46:14,Z39.50 Query Error,Connection lost,z3950.library.columbia.edu:7090,Source\biblioteq_book.cc,4656
2021/09/27 17:46:09,Z39.50 Query Error,Connect failed,opac.bnrm.ma:1111,Source\biblioteq_book.cc,4656
2021/09/27 17:46:51,Z39.50 Query Error,Z39.50 Empty Results Set,Z39.50 Empty Results Set,Source\biblioteq_book.cc,4656

**Comment by textbrowser** [comment]
So the first two causes are clear. Perhaps the Z39.50 servers are preventing access for international customers. The third indicates that the query resulted in zero results.

**Comment by NetCzech** [comment]
I'm looking for the[ same book](https://mitpress.mit.edu/books/quantum-computing-everyone) as you. Do you think there is a problem in the country I am looking for?

BTW Settings for the National Library of Czech Republic are correct?

**Comment by textbrowser** [comment]
I cannot tell which server you queried from the third entry. Perhaps I should include more information in the results. I assume that the customer provides such information as it's very useful. :) However, I remind myself that it's best to include the information so that it's more open for everyone. It isn't a free implementation. Anyway, of course. I do not believe that Z39.50 are open to everyone. If it works here, it is not guaranteed to work elsewhere. The services are free for some, costly for others.

**Comment by NetCzech** [comment]
I tried different connections to libraries from [NTK](https://www.techlib.cz/cs/83296-z39-50-targets). Fortunately unfortunately none work. I'm not sure all resources will be blocked.

I also found a description of the problem [here](https://sourceforge.net/p/biblioteq/discussion/846456/thread/0834e940/), I do not have an entry in the registry (portable version).

Setting in conf file is ok? 

Probably an unsolvable problem. Thank you.

**Comment by textbrowser** [comment]
Will try some of the NTK libraries. In which way do they fail?

**Comment by textbrowser** [comment]
New record_syntax.

https://github.com/textbrowser/biblioteq/commit/e866022affe76b5e246576c698aa53f3af1fc0eb

**Comment by textbrowser** [comment]
Your entry is not correct. The following definition is correct.

database_name = skc-utf
format = "utf-8"
hostname = aleph.nkp.cz
name = National Library of the Czech Republic
password =
port = 9991
proxy_host =
proxy_port =
record_syntax = "MARC21"
username =

**Comment by NetCzech** [comment]
I tried 6 books and the search worked well.
Thank you very much for your help.

**Comment by textbrowser** [comment]
:)

**Comment by textbrowser** [comment]
So I do not have to try your libraries. Awesome.
